### PR TITLE
GHAE registration flow - updated UI & errors

### PR DESCRIPTION
**Pull request by ankitkaushal01**
this pr resolves registration flow for jira app with github ae.

closes https://github.com/github/marketplace/issues/2085

### CODE REVIEW

File: lib/frontend/app.js [hunk 1]
```diff
@@ -121,7 +122,9 @@ module.exports = (appTokenGenerator) => {
     res.redirect(info.external_url);
   });
 
-  app.get('/ghaeRegister', async (req, res, next) => res.render('ghae_register.hbs'));
+  app.get('/ghaeRegister', async (req, res, next) => res.render('ghae_register.hbs', {
```
Comment (chshrikh) [inline lib/frontend/app.js]: change the routing url to GitHubAEReigister

File: lib/frontend/post-ghae-complete.js [hunk 2]
```diff
@@ -22,38 +22,44 @@ module.exports = function (opts) {
 
     // check valid referer, needed to validate ghaehost
     if (!req.get('referer')) {
-      handleExceptions(res, 400, 'Request not coming from valid referer');
-      return;
+      return res.status(400).render('ghae-register-error.hbs', {
+        error: 'Request not coming from valid referer',
```
Comment (chshrikh) [inline lib/frontend/post-ghae-complete.js]: can we add trace here, we need to trace the source request IP. For now we can use the existing logger base utility

File: lib/util/constants.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const GHAE_URL_PATTERN_PROD = 'ghe.com';
+const GHAE_URL_PATTERN_DEV = 'ghaekube.net';
+const GHAE_INSTANCE_URL_DEV = 'https://ghaebuild48746.ghaekube.net';
```
Comment (chshrikh) [inline lib/util/constants.js]: change 'https://ghaebuild48746.ghaekube.net' to "https://GhitHubAEAccount.ghaekube.net"

File: lib/frontend/post-ghae-register.js [hunk 4]
```diff
@@ -70,8 +71,7 @@ module.exports = function (opts) {
 
   function validateGhaehost(ghaeHost) {
     // TODO: validate ghae url based on certificates
-    let ghaeUrlPattern = 'ghaekube.net';
-    if (process.env.NODE_ENV === 'production') ghaeUrlPattern = 'ghe.com';
+    let ghaeUrlPattern = constants.GHAE_URL_PATTERN;
     if (ghaeHost.includes(ghaeUrlPattern)) return true;
```
Comment (chshrikh) [inline lib/frontend/post-ghae-register.js]: make it endWith

**Review by chshrikh — APPROVED** [review]
take care of couple of comments, looks good to me

File: lib/frontend/post-ghae-complete.js [hunk 5]
```diff
@@ -22,38 +22,49 @@ module.exports = function (opts) {
...
+        error: 'Invalid Auth state parameter',
+      });
     }
     let ghaeHost = parseHost(req.get('referer'));
     let githubHost = registration.githubHost;
     // check valid ghaehost
     if (githubHost !== ghaeHost) {
-      handleExceptions(res, 401, 'Request coming from invalid host');
-      return;
+      console.log(`Request not coming from valid GitHub AE instance ${JSON.stringify(query)} - request ip ${req.ip}`);
```
Comment (AmrutaKawade) [inline lib/frontend/post-ghae-complete.js]: can you replace console.log with 

```
const { getLog } = require('../../lib/config/logger');
let logger = getLog();
logger.error
```

File: script/db_create [hunk 6]
```diff
@@ -6,10 +6,4 @@ set -e
 
 PATH="$PATH:node_modules/.bin"
 
-# dev
```
Comment (AmrutaKawade) [inline script/db_create]: don't we need this?

**Review by AmrutaKawade — APPROVED** [review]

**Pull request merged**